File: android2_lesson01/src/java/java/com/example/ama/android2_lesson01/ui/rv/adapter/ListOfNotesAdapter.java
package com.example.ama.android2_lesson01.ui.rv.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.ama.android2_lesson01.R;
import com.example.ama.android2_lesson01.model.Note;
import com.example.ama.android2_lesson01.ui.rv.ListOfNotesFragment;

import java.util.ArrayList;

/**
 * Adapter class for RecyclerView of {@link ListOfNotesFragment}
 *
 * @see ListOfNotesFragment
 */

public class ListOfNotesAdapter extends RecyclerView.Adapter<ListOfNotesHolder> {

    private ArrayList<Note> mData = new ArrayList<>();
    private ListOfNotesHolder.OnNoteClickListener mListener;

    public ListOfNotesAdapter(ListOfNotesHolder.OnNoteClickListener listener) {
        mListener = listener;
    }

    @NonNull
    @Override
    public ListOfNotesHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_rv, parent, false);
        return new ListOfNotesHolder(itemView, this, mListener);
    }

    @Override
    public void onBindViewHolder(@NonNull ListOfNotesHolder holder, int position) {
        holder.bind(mData.get(position));
    }

    @Override
    public int getItemCount() {
        return mData.size();
    }

    /**
     * Get ArrayList of all displaying {@link Note} objects
     *
     * @return ArrayList of notes
     * @see Note
     */

    public ArrayList<Note> getmData() {
        return mData;
    }

    /**
     * Set ArrayList of all displaying {@link Note} objects and notify adapter about changes
     *
     * @param mData ArrayList of notes
     */

    public void setmData(ArrayList<Note> mData) {
        this.mData = mData;
        notifyDataSetChanged();
    }
}

File: android2_lesson01/src/kotlin/java/com/example/ama/android2_lesson01/ui/rv/adapter/ListOfNotesAdapter.kt
package com.example.ama.android2_lesson01.ui.rv.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.ama.android2_lesson01.R
import com.example.ama.android2_lesson01.model.Note
import kotlinx.android.synthetic.main.item_rv.view.*

/**
 * Adapter class for RecyclerView of [com.example.ama.android2_lesson01.ui.rv.ListOfNotesFragment]
 *
 * @see com.example.ama.android2_lesson01.ui.rv.ListOfNotesFragment
 */

class ListOfNotesAdapter(
        private val deleteListener: (Note) -> Unit,
        private val editListener: (Note) -> Unit)
    : RecyclerView.Adapter<ListOfNotesAdapter.ListOfNotesHolder>() {

    private var mData: ArrayList<Note> = ArrayList()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ListOfNotesHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.item_rv, parent, false)
        return ListOfNotesHolder(itemView)
    }

    override fun onBindViewHolder(holder: ListOfNotesHolder, position: Int) {
        holder.bind(mData[position])
    }

    override fun getItemCount(): Int = mData.size

    /**
     * Set ArrayList of all displaying [Note] objects and notify adapter about changes
     *
     * @param mData ArrayList of notes
     */

    fun setmData(mData: ArrayList<Note>) {
        this.mData = mData
        notifyDataSetChanged()
    }

    /**
     * Class of holder for RecyclerView from [com.example.ama.android2_lesson01.ui.rv.ListOfNotesFragment]
     *
     * @see com.example.ama.android2_lesson01.ui.rv.ListOfNotesFragment
     */

    inner class ListOfNotesHolder(itemView: View)
        : RecyclerView.ViewHolder(itemView) {

        fun bind(note: Note) {
            itemView.tv_card_note_title.text = note.title
            itemView.tv_card_note_text.text = note.text
            itemView.ibtn_delete_note.setOnClickListener { deleteListener.invoke(note) }
            itemView.cv_note.setOnClickListener { editListener.invoke(note) }
        }
    }
}
